src/main/java/app.java

```diff
@@ -6,15 +6,23 @@ public static void main(String[] args) {
         app main = new app();
         File file = new File(main.getClass().getClassLoader().getResource("Z:\\Downloads\\Project%20Workspace\\Practice%20workspace\\codegur\\sample1\\src\\main\\resources\\data.txt").getFile());
         System.out.println(file.exists());
-    try {
-        FileReader reader = new FileReader(file);
-        BufferedReader br = new BufferedReader(reader);
-        String line;
-        while ((line = br.readLine()) != null) {
-            System.out.println(line);
+        File file1 = new File(main.getClass().getClassLoader().getResource("Z:\\Downloads\\Project%20Workspace\\Practice%20workspace\\codegur\\sample1\\src\\main\\resources\\data1.txt").getFile());
+        try {
+            FileReader reader = new FileReader(file);
+            BufferedReader br = new BufferedReader(reader);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *br*. Currently, there are execution paths that do not contain closure statements. Either a) close *br* in a try-finally block or b) close the resource by declaring *br* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).